src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+        }
+    
+        JListFrame selectProblemsFrame = new JListFrame("Select Problems ", problems, selectedProblemsIndexes, new ISelectedListsSetter() {
+            
+            @Override
+            public void setSelectedValuesList(List<Object> selectedValuesList, int[] selectedIndices) {
+                selectedProblemsIndexes = selectedIndices;
+                updateProblemsLabel(selectedValuesList);
+            }
+        });
+        selectProblemsFrame.setVisible(true);
+    }
+
+    protected void updateCDPDirAndFields(String cdpConfigDir) {
+	    try {
+            File configFile = new File(cdpConfigDir);
+            String cdpRootDirectory = configFile.getAbsolutePath();
+//            if (configFile.getAbsolutePath().endsWith(IContestLoader.CONFIG_DIRNAME)) {
+//                cdpRootDirectory = configFile.getParent();
+//            }
+	        
+	        cdpTextField.setText(cdpRootDirectory);
+	        updateClientCDPPath(cdpRootDirectory);
+	        updateNosLabel(null);
+        } catch (Exception e) {
+            log.log(Level.WARNING, "Problem updating CDP Dir", e);
+        }
+    }
+
+    /**
+	 * Dialog with question mark ison for info on selecting judgement types.
+	 * @return
+	 */
+	private JLabel getWhatsThisOne() {
+		
+		String[] messageLines = {
+				"Selecting judgement types", //
+				"", //
+				"Use Select button to select judgement types", //
+		};
+		
+		return FrameUtilities.getWhatsThisLabel("Selecting judgement samples", messageLines);
+	}
+	
+	/**
+	 * Repopulate submissionSolutionList.
+	 * 
+	 * @return 
+	 */
+	public SubmissionSolutionList getSubmissionSolutionList() {
+		String cdpPath = cdpTextField.getText();
+		
+		SubmissionSolutionList list = new SubmissionSolutionList(new File(cdpPath));
+		Collections.reverse(list);
+		submissionSolutionList = list;
+		
+		return submissionSolutionList;
+	}
+
+	protected void showJudgementTypeList() {
+		
+		if (getSubmissionSolutionList() == null || getSubmissionSolutionList().size() == 0) {
+			showMessage("No submission judgement types found in dir: "+cdpTextField.getText());
+			return;
+		}
+		
+		SubmissionSampleLocation[] listData = toArray(getSubmissionSolutionList());
+	
+		JListFrame selectNoSolutionsFrame = new JListFrame("Select No/Failed Judges solutions", listData, selectedJudgingTypeIndexes, new ISelectedListsSetter() {
+			
+			@Override
+			public void setSelectedValuesList(List<Object> selectedValuesList, int[] selectedIndices) {
+				selectedJudgingTypeIndexes = selectedIndices;
+				updateNosLabel(selectedValuesList);
+			}
+		});
+		selectNoSolutionsFrame.setVisible(true);
+	}
+
+	private SubmissionSampleLocation[] toArray(SubmissionSolutionList list) {
+		return (SubmissionSampleLocation[]) list.toArray(new SubmissionSampleLocation[list.size()]);
+	}
+	
+	/**
+	 * Update selected languages label and selected language list
+	 * @param valuesList
+	 */
+	protected void updateLanguagesLabel(List<Object> valuesList) {
```

Comment:
As noted above, I suggest surrounding the message ( "xxx selected") with parentheses, so it looks like either
`(None selected)` or `(3 selected)`.